lib/client/api.go

```diff
@@ -1976,15 +1989,14 @@ func (tc *TeleportClient) Login(ctx context.Context) (*Key, error) {
 	key.ProxyHost = webProxyHost
 	key.TrustedCA = response.HostSigners
 
-	// Check that a host certificate for at least one cluster was returned and
-	// extract the name of the current cluster from the first host certificate.
-	if len(response.HostSigners) <= 0 {
-		return nil, trace.BadParameter("bad response from the server: expected at least one certificate, got 0")
+	// Store the requested cluster name in the key.
+	key.ClusterName = tc.SiteName
+	if key.ClusterName == "" {
+		rootClusterName := key.TrustedCA[0].ClusterName
```

Comment:
Both amount to the same thing because of the `key.TrustedCA = response.HostSigners` right above. I just thought inferring the root cluster name from `TrustedCA` would be more self-explanatory to readers.